src/python/grpcio/tests/qps/worker_server.py

```diff
@@ -0,0 +1,179 @@
+    return control_pb2.ServerStatus(stats=stats, port=port, cores=cores)
+
+  def _create_server(self, config):
+    if config.server_type == control_pb2.SYNC_SERVER:
+      servicer = benchmark_server.BenchmarkServer()
+      server = services_pb2.beta_create_BenchmarkService_server(servicer)
+    elif config.server_type == control_pb2.ASYNC_GENERIC_SERVER:
+      resp_size = config.payload_config.bytebuf_params.resp_size
+      servicer = benchmark_server.GenericBenchmarkServer(resp_size)
+      method_implementations = {
+          ('grpc.testing.BenchmarkService', 'StreamingCall'):
+          utilities.stream_stream_inline(servicer.StreamingCall),
+          ('grpc.testing.BenchmarkService', 'UnaryCall'):
+          utilities.unary_unary_inline(servicer.UnaryCall),
+      }
+      server = implementations.server(method_implementations)
+    else:
+      raise Exception('Unsupported server type {}'.format(config.ServerType))
+
+    if config.HasField('security_params'):  # Use SSL
+      server_creds = implementations.ssl_server_credentials([(
+          resources.private_key(), resources.certificate_chain())])
+      port = server.add_secure_port('[::]:{}'.format(config.port), server_creds)
+    else:
+      port = server.add_insecure_port('[::]:{}'.format(config.port))
+
+    return (server, port)
+
+  def RunClient(self, request_iterator, context):
+    config = next(request_iterator).setup
+    client_runners = []
+    qps_data = histogram.Histogram(config.histogram_params.resolution,
+                                   config.histogram_params.max_possible)
+    start_time = time.time()
+
+    # Create a client for each channel
+    for i in xrange(config.client_channels):
+      server = config.server_targets[i % len(config.server_targets)]
+      runner = self._create_client_runner(server, config, qps_data)
+      client_runners.append(runner)
+      runner.start()
+
+    end_time = time.time()
+    yield self._get_client_status(start_time, end_time, qps_data)
+
+    # Respond to stat requests
+    for request in request_iterator:
+      end_time = time.time()
+      status = self._get_client_status(start_time, end_time, qps_data)
+      if request.mark.reset:
+        qps_data.reset()
+        start_time = time.time()
+      yield status
+
+    # Cleanup the clients
+    for runner in client_runners:
+      runner.stop()
+
+  def _get_client_status(self, start_time, end_time, qps_data):
+    latencies = qps_data.get_data()
+    end_time = time.time()
+    elapsed_time = end_time - start_time
+    stats = stats_pb2.ClientStats(latencies=latencies,
+                                  time_elapsed=elapsed_time,
+                                  time_user=elapsed_time,
+                                  time_system=elapsed_time)
+    return control_pb2.ClientStatus(stats=stats)
+
+  def _create_client_runner(self, server, config, qps_data):
+    if config.client_type == control_pb2.SYNC_CLIENT:
```

Comment:
SYNC_CLIENT is not the only possible value of the enum, so you should probably handle this more explictly. (what you have above for server looks fine `raise Exception('Unsupported server type {}'.format(config.ServerType))`)